R/E_loo.R

```diff
@@ -199,12 +193,11 @@ E_loo.matrix <-
 .wmean <- function(x, w, ...) {
   sum(w * x)
 }
-.wvar <- function(x, w, r_eff = NULL, ...) {
-  if (is.null(r_eff)) {
-    r_eff <- 1
-  }
-  r <- (x - .wmean(x, w))^2
-  sum(w^2 * r) / r_eff
+.wvar <- function(x, w, ...) {
+  (sum(.wmean(x^2, w)) - sum(.wmean(x, w)^2)) / (1 - 1/length(w))
```

Comment:
While thinking a test with non-equal w's, I realized that here `length(w)` should be replaced with the effective sample size `1/sum(w^2)`. To make it clear, maybe write
```
ess <- 1/sum(w.^2);
(sum(.wmean(x^2, w)) - sum(.wmean(x, w)^2)) / (1 - 1/ess)
```

To see why we need to use ess, consider `w <- c(rep(0.1,10), rep(0, 90))`, which has 10 non-zero but equal weights, and the correct variance would be `var(x[w>0])`. Using the ESS estimate `1/sum(w^2)` works correctly then also when all weights are non-zero bu not equal.

Reply:
I also first thought proposing `(1- sum(w^2))` but maybe i can be used with a comment like
```
# (1- sum(w^2)) is equal to (ESS-1)/ESS, where effective sample size ESS is estimated with the generic target quantity invariant estimate 1/sum(w^2), see e.g. "Monte Carlo theory, methods and examples" by Owen (2013).

Reply:
Ok the `.wvar` function now has denominator `(1 - sum(w^2))` and the comment you suggested adding. Can you take one more quick look and then if everything looks good we can merge this. 

I will also run reverse dependency checks to make sure that none of the packages that depend on loo have errors resulting from this change (for example if they have a test in their package using `E_loo(type="var")` then the result will change). If there are errors I will notify maintainers of the upcoming change.

Reply:
Looks good now, and I also tested with a slightly different example that the computation matches var

Reply:
> I will also run reverse dependency checks to make sure that none of the packages that depend on loo have errors resulting from this change (for example if they have a test in their package using E_loo(type="var") then the result will change).

No issues with reverse dependencies